core/src/main/scala/kafka/server/DynamicConfig.scala

```diff
@@ -65,6 +65,12 @@ object DynamicConfig {
     def names = brokerConfigDef.names
 
     def validate(props: Properties) = DynamicConfig.validate(brokerConfigDef, props, customPropsAllowed = true)
+
+    def typeOf(key: String): ConfigDef.Type =
+      brokerConfigDef.configKeys.get(key) match {
+        case null => null
```

Comment:
nit: Changing `configKeys` to scala list can change this whole code to:

> brokerConfigDef.configKeys.asScala.get(key).map(_.`type`).orNull

This has benefit that we can get rid of null case and don't have to be careful to write `case null` before any other case.

Reply:
One day I'll get this scala magic down... done.